module/VuFind/src/VuFind/Controller/AlphabrowseController.php

```diff
@@ -53,50 +106,15 @@ public function homeAction()
     {
         $config = $this->getConfig();
 
-        // Load browse types from config file, or use defaults if unavailable:
-        if (isset($config->AlphaBrowse_Types)
-            && !empty($config->AlphaBrowse_Types)
-        ) {
-            $types = [];
-            foreach ($config->AlphaBrowse_Types as $key => $value) {
-                $types[$key] = $value;
-            }
-        } else {
-            $types = [
-                'topic'  => 'By Topic',
-                'author' => 'By Author',
-                'title'  => 'By Title',
-                'lcc'    => 'By Call Number'
-            ];
-        }
-
-        // Load any extras from config file
-        $extras = [];
-        if (isset($config->AlphaBrowse_Extras)) {
-            foreach ($config->AlphaBrowse_Extras as $key => $value) {
-                $extras[$key] = $value;
-            }
-        } else {
-            $extras = [
-                'title' => 'author:format:publishDate',
-                'lcc' => 'title',
-                'dewey' => 'title'
-            ];
-        }
-
-        // Load remaining config parameters
-        $rows_before = isset($config->AlphaBrowse->rows_before)
-            && is_numeric($config->AlphaBrowse->rows_before)
+        // Load config parameters
+        $types = $this->getTypes($config);
+        $extras = $this->getExtras($config);
+        $rows_before = is_numeric($config->AlphaBrowse->rows_before ?? null)
             ? (int)$config->AlphaBrowse->rows_before : 0;
         $highlighting = $config->AlphaBrowse->highlighting ?? false;
-        $limit  = isset($config->AlphaBrowse->page_size)
-            && is_numeric($config->AlphaBrowse->page_size)
+        $limit  = is_numeric($config->AlphaBrowse->page_size ?? null)
```

Comment:
Thanks for the suggestion; I've adjusted this accordingly.

Reply:
Do you know if $config always returns the value as a string? If not, it would be safest to cast to string before checking with ctype_digit, as it would return false for an integer.

Reply:
I addressed this in #1977.